packages/orbit-components/src/HorizontalScroll/index.tsx

```diff
@@ -226,54 +108,65 @@ const HorizontalScroll = React.forwardRef<HTMLDivElement, Props>(
     }, [handleOverflow, handleResize]);
 
     return (
-      <StyledWrapper
-        {...props}
-        $minHeight={minHeight}
-        overflowElevation={overflowElevation}
+      <div
+        className={cx(
+          "relative inline-flex w-full items-center overflow-hidden",
+          isOverflowing && (isDragging ? "cursor-grabbing" : "cursor-grab"),
+        )}
         data-test={dataTest}
         id={id}
-        isDragging={isDragging}
-        isEnd={reachedEnd}
-        isStart={reachedStart}
-        isOverflowing={isOverflowing}
         ref={mergeRefs<HTMLDivElement>([ref, containerRef])}
-        elevationColor={theme.orbit[elevationColor]}
+        style={{ minHeight }}
       >
+        {overflowElevation && (
+          <>
+            <ElevationHaze
+              className={cx("left-0", (!isOverflowing || reachedStart) && "invisible")}
+              style={{ boxShadow: `5px 0px 20px 20px ${theme.orbit[elevationColor]}` }}
+            />
+            <ElevationHaze
+              className={cx("right-0", (!isOverflowing || reachedEnd) && "invisible")}
+              style={{ boxShadow: `-5px 0px 20px 20px ${theme.orbit[elevationColor]}` }}
```

Comment:
My pleasure 🙂

Reply:
Also, to note: I’ve removed emojis from the visual story since their rendering were unstable across runs.

![visual-HorizontalScroll-scroll-1-diff](https://github.com/kiwicom/orbit/assets/821851/93481596-29a9-48e4-9226-83d62cd56e3a)